src/analysis/processing/qgsalgorithmuniquevalues.cpp

```diff
@@ -0,0 +1,198 @@
+/***************************************************************************
+                         qgsalgorithmuniquevalues.cpp
+                         ---------------------
+    begin                : May 2025
+    copyright            : (C) 2025 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmuniquevalues.h"
+
+///@cond PRIVATE
+
+QString QgsUniqueValuesAlgorithm::name() const
+{
+  return QStringLiteral( "listuniquevalues" );
+}
+
+QString QgsUniqueValuesAlgorithm::displayName() const
+{
+  return QObject::tr( "List unique values" );
+}
+
+QStringList QgsUniqueValuesAlgorithm::tags() const
+{
+  return QObject::tr( "count,unique,values" ).split( ',' );
+}
+
+QString QgsUniqueValuesAlgorithm::group() const
+{
+  return QObject::tr( "Vector analysis" );
+}
+
+QString QgsUniqueValuesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectoranalysis" );
+}
+
+QString QgsUniqueValuesAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Returns list of unique values in given field(s) of a vector layer." );
+}
+
+QString QgsUniqueValuesAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Returns list of unique values in given field(s) of a vector layer." );
+}
+
+QgsUniqueValuesAlgorithm *QgsUniqueValuesAlgorithm::createInstance() const
+{
+  return new QgsUniqueValuesAlgorithm();
+}
+
+void QgsUniqueValuesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ) ) );
```

Comment:
Need to make sure this accepts geometryless layers too:
```suggestion
  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ), QList<int>() << static_cast<int>( Qgis::ProcessingSourceType::Vector ) ) );
```